### DISCUSSION TITLE
Pin Multiplexing for both versions | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#580] [Q&A] [answered]
# Pin Multiplexing for both versions | Seeed Studio Wiki

Pin Multiplexing on Seeed Studio XIAO nRF52840 (Sense)

https://wiki.seeedstudio.com/XIAO-BLE-Sense-Pin-Multiplexing/

**Comment by kjkent** [comment]
The I2C guide here shows no external pullups connected on SDA or SCL -- does the XIAO nRF52840 Sense have internal pullups? 

I'm trying to use I2C on CircuitPython, but I'm receiving the error:
```Python
Adafruit CircuitPython 8.2.4 on 2023-08-22; Seeed XIAO nRF52840 Sense with nRF52840
>>> import time
>>> import board
>>> i2c = board.I2C()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: No pull up found on SDA or SCL; check your wiring
```

**Reply by xzxcessarr** [reply, marked as answer]
Hello, you can use this way to set your I2C pins at software level by the bitbangio library :
`import board
import bitbangio
I2C=bitbangio.I2C(board.A1, board.A2)`

**Comment by rkvaw** [comment]
Could we redefine SPI to other Pins of the Xiao nRF52840 pins 
for example : SDI -> pin 3, SDO -> pin4, SCK -> pin5, CS -> pin9
